orangecontrib/text/widgets/tests/test_concordances.py

```diff
@@ -57,6 +57,34 @@ def test_data_non_displayroles(self):
         self.assertIsInstance(model.data(ind00, Qt.TextAlignmentRole),
                               (Qt.Alignment, type(None)))
 
+    def test_color_proper_rows(self):
+        """Rows are colored corresponding to the document"""
+        model = ConcordanceModel()
+        model.set_width(2)
+        model.set_corpus(self.corpus)
+        model.set_word("of")
+
+        self.assertIsInstance(model.data(model.index(0, 0), Qt.BackgroundRole),
+                              (QColor, (255, 255, 255)))
```

Comment:
`assertIsInstance(obj, type)` checks whether `obj` is an instance of `type`, where type can be a type or a tuple of types. Here, you only test that what the model returns is an instance of `QColor` or ... something that is not a type, but is fortunately not checked since the first argument, `QColor`, already matches.

What you wanted to check was

    self.assertEqual(model.data(model.index(0, 0), Qt.BackgroundRole),
                     QColor(255, 255, 255))

However, this is not what you should want the test to check. The model doesn't have to produce exactly these two colors, `(255, 255, 255)` and `(236, 243, 255)`. Any two different colors will do. So you want to check that rows 0 and 1 have the same color, and that row 2 has a different color.

I would thus formulate the test as

        color1 = model.data(model.index(0, 0), Qt.BackgroundRole)
        self.assertEqual(model.data(model.index(1, 0), Qt.BackgroundRole),
                         color1)
        self.assertNotEqual(model.data(model.index(2, 0), Qt.BackgroundRole),
                            color1)

And that'd be it. You can check the colors in other columns, too, but returning different colors for different columns is such an unlikely bug that I'd omit this for the sake of space and clarity. Your choice, though.